### PR TITLE
Add tagging to CF template and to serverless resources

### PR DESCRIPTION
**Pull request by caggle**

### CODE REVIEW

**Comment by caggle** [comment]
In order to tag everything consistently - every resource that supports tag, created by the CI pipeline and Serverless framework, I am proposing this:

- Add a new parameter to the CF template, called `ServiceTag` with a default value of "MozDef-EF"
- Construct the actual tag value as <ServiceTag>-<Environment> (for instance https://github.com/mozilla/mozdef-event-framework/blob/add_tagging/templates/codepipeline-cf-template.yml#L111)
- Add this tag to all resources supported in this template.
- In order for the serverless resources to have the same tags, add `stackTags` to `serverless.yml` (https://github.com/mozilla/mozdef-event-framework/blob/add_tagging/serverless.yml#L53)
- This adds the tags to _most_ resources, not all (I am not sure why), for instance SQS queues are not tagged this way. In order to forcefully tag them as well, add `Tags` to the resources section of `serverless.yml` (like https://github.com/mozilla/mozdef-event-framework/blob/add_tagging/serverless.yml#L136)

The whole thing makes a breaking change however, due to this: https://github.com/mozilla/mozdef-event-framework/blob/add_tagging/serverless.yml#L54, i.e. I am proposing to change the value of the `project`variable from "mozdef-event-framework" - which was hardcoded - to an environment variable, coming from buildspec. This means:

- We will need to update the buildspec to declare another env variable called PROJECT, with the format "MozDef-EF-STAGE (e.g. MozDef-EF-dev)
- We will need to change the SSM parameter path for auth token, as in serverless it is currently using the value of the `project` variable (https://github.com/mozilla/mozdef-event-framework/blob/add_tagging/serverless.yml#L83), which I am proposing to change here.

Apologies if this sounds too complex, we don't have to implement this. I just think this would be another step to make the whole thing uniform and generic. 

Apart from merging this PR, the 2 steps above would need to be done one-time manually, so it shouldn't be a chore every time.

**Comment by caggle** [comment]
> - Add a new parameter to the CF template, called ServiceTag with a default value of "MozDef-EF"

Actually, if we are happy with the tagging convention of "MozDeF-EF-STAGE" (so stuff would be tagged like "MozDef-EF-dev" for instance) for resources for each stack we deploy, we don't even need a user controlled parameter "ServiceTag" as above. We can just hardcode this in the template. 

So for instance, [this line](https://github.com/mozilla/mozdef-event-framework/blob/add_tagging/templates/codepipeline-cf-template.yml#L111) would instead read:
`Value: !Join [ "", ["MozDef-EF", "-", !Ref 'Environment' ] ]`

@Phrozyn please let me know what you think.

**Comment by Phrozyn** [comment]
I'm good with the tagging you proposed in the last comment.
I'm also fine with removing mozdef-event-framework as the hard coded project name. Makes sense. I'll wait till we can discuss before we merge.

**Review by Phrozyn — APPROVED** [review]

**Pull request merged**
